src/SFML/Graphics/Texture.cpp

```diff
@@ -198,7 +198,7 @@ bool Texture::create(const Vector2u& size)
     static const bool textureEdgeClamp = GLEXT_texture_edge_clamp || GLEXT_GL_VERSION_1_2 ||
                                          Context::isExtensionAvailable("GL_EXT_texture_edge_clamp");
 
-    if (!m_isRepeated && !textureEdgeClamp)
+    if (!texture.m_isRepeated && !textureEdgeClamp)
```

Comment:
Ditto for the other checks -- unless I am missing something, they should all have their default value, as this is a brand new texture. Right?

Reply:
`m_isRepeated` defaults to `false` and is never touched prior to this line so yeah that will always evaluate to `true` and thus can be removed.

I think this whole section can be removed as well.
```cpp
    if (texture.m_sRgb && !textureSrgb)
    {
        static bool warned = false;

        if (!warned)
        {
#ifndef SFML_OPENGL_ES
            err() << "OpenGL extension EXT_texture_sRGB unavailable" << '\n';
#else
            err() << "OpenGL ES extension EXT_sRGB unavailable" << '\n';
#endif
            err() << "Automatic sRGB to linear conversion disabled" << std::endl;

            warned = true;
        }

        texture.m_sRgb = false;
    }
    ```
`m_sRgb` defaults to `false` and is never touched prior to this line so I don't think there is ever a way to enter this conditional. This is a good sign because it shows that this API change systematically eliminates certain errors that were made possible by the previous API.

Reply:
As I look closer, `Texture::setSrgb` is only useful if you call it prior to loading the texture. I think that needs to be added as a function parameter to ensure there is still a way to set this. I believe the same can be said for `Texture::setSmooth` and `Texture::setRepeated`.